src/optree.cpp

```diff
@@ -26,6 +26,16 @@ limitations under the License.
 #include <pybind11/pybind11.h>
 #include <pybind11/stl.h>
 
+#if defined(PYBIND11_HAS_NATIVE_ENUM) ||                                                           \
```

Comment:
Sorry it's this complicated: We should change pybind11, to move `#define PYBIND11_HAS_NATIVE_ENUM` into pybind11/detail/common.h

Is there a chance that you can send a PR for that?

Reply:
Actually, please don't worry about the PR.

I just looked more, we should also move `PYBIND11_HAS_INTERNALS_WITH_SMART_HOLDER_SUPPORT` to common.h. I'll move both in one PR. I'll take care of it soon (before the v3.0.0 release).